### PR TITLE
Porting changes from 0.9.6 to 1.0.0

### LINKED ISSUE #743: Port changes from 0.9.6 to 1.0.0
Port block verification and `onReceiveBlock` changes from `0.9.6` to `1.0.0`. Including unit tests.

### LINKED ISSUE #754: I can not visit the webpage ran by docker

### PR DESCRIPTION
**Pull request by karmacoma**
- Key commits are: 7b474a2 and c954203
- Added some pending examples in ff05db2
- Other commits were for beautification and standardisation purposes

Apart from pending examples addition, all work from `0.9.6`.

Closes #743

### CODE REVIEW

**Comment by karmacoma** [comment]
Branch name should have been: `743-0.9.6-port`. Will leave as is.

File: modules/blocks/verify.js [hunk 1]
```diff
@@ -89,6 +89,341 @@ __private.checkTransaction = function (block, transaction, cb) {
...
+ * Verify block before processing and return all possible errors related to block
+ *
+ * @public
+ * @method verifyBlock
+ * @param  {Object}  block Full block
+ * @return {Object}  result Verification results
+ * @return {boolean} result.verified Indicator that verification passed
+ * @return {Array}   result.errors Array of validation errors
+ */
+Verify.prototype.verifyBlock = function (block) {
```
Comment (LucasIsasmendi) [inline modules/blocks/verify.js]: `verifyBlock` function was previously defined

Comment (karmacoma) [inline modules/blocks/verify.js]: Thanks @LucasIsasmendi

File: modules/blocks/verify.js [hunk 2]
```diff
@@ -89,6 +89,307 @@ __private.checkTransaction = function (block, transaction, cb) {
...
+	result = __private.verifyReward(block, result);
+	result = __private.verifyId(block, result);
+	result = __private.verifyPayload(block, result);
+
+	result.verified = result.errors.length === 0;
+	result.errors.reverse();
+
+	return result;
+};
+/**
```
Comment (diego-G) [inline modules/blocks/verify.js]: Add CR

**Comment by diego-G** [comment]
I got an error when executing `test/modules/blocks/process.js` locally:
```[dbg] 2017-09-01 10:17:40 | End broadcast
[log] 2017-09-01 10:17:40 | error - error: ERROR:  insert or update on table "blocks" violates foreign key constraint "blocks_previousBlock_fkey"
DETAIL:  Key (previousBlock)=(10284725956204223183) is not present in table "blocks".
[log] 2017-09-01 10:17:40 | error - tx: INSERT INTO "blocks"("id","version","timestamp","height","previousBlock","numberOfTransactions","totalAmount","totalFee","reward","payloadLength","payloadHash","generatorPublicKey","blockSignature") VALUES ('4343586543742503802',0,40151860,11,'10284725956204223183',0,0,0,0,0,'\xe3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855','\x910da2a8e20f25ccbcb029fdcafd369b43d75e5bc4dc6d92352c29404acc350f','\x92916860b00c5271f5cccff02544143a4ace1cb7630b4658f189685e75e0ad00d3baa21512bdd1dcfb0bee154531b35e80105a6246daa632bfd4bf77d7fc1b02')
(node:60978) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 8): BatchError {
    stat: { total: 1, succeeded: 0, failed: 1, duration: 0 }
    errors: [
        0: { Error: ERROR:  insert or update on table "blocks" violates foreign key constraint "blocks_previousBlock_fkey"
        DETAIL:  Key (previousBlock)=(10284725956204223183) is not present in table "blocks".
        
            at Client._readError (/Users/diego-g/git/lisk/node_modules/pg-native/index.js:80:13)
            at Client._read (/Users/diego-g/git/lisk/node_modules/pg-native/index.js:121:19)
            at emitNone (events.js:86:13)
            at PQ.emit (events.js:185:7)
            at PQ.EventEmitter.emit (/Users/diego-g/git/lisk/node_modules/sc-domain/index.js:12:31)
          severity: 'ERROR',
          sqlState: '23503',
          messagePrimary: 'insert or update on table "blocks" violates foreign key constraint "blocks_previousBlock_fkey"',
          messageDetail: 'Key (previousBlock)=(10284725956204223183) is not present in table "blocks".',
          sourceFile: 'ri_triggers.c',
          sourceLine: '3324',
          sourceFunction: 'ri_ReportViolation' }
    ]
}```

More with `test/unit/modules/app.js`.

I've launched Jenkins to confirm it.

**Comment by MaciejBaj** [comment]
*rounds

**Comment by MaciejBaj** [comment]
@diego-G the error is caused by running blocks/process.js tests before - it applies and then removes blocks to DB which makes then impossible to create a new one after running it. It has nothing to do with this PR and will be fixed in #748

File: modules/blocks/process.js [hunk 3]
```diff
@@ -47,6 +47,148 @@ function Process (logger, block, peers, transaction, schema, db, dbSequence, seq
 }
 
 /**
+ * Receive block - logs info about received block, updates last receipt, processes block
```
Comment (MaciejBaj) [inline modules/blocks/process.js]: ??

Comment (diego-G) [reply modules/blocks/process.js]: Ordering prototypes at the bottom and __private functions at the top.

File: test/api/peers.js [hunk 4]
```diff
@@ -483,7 +478,7 @@ describe('GET /api/peers', function () {
 	});
 });
 
-describe('GET /api/peers/get', function () {
+describe.skip('GET /api/peers/get', function () {
```
Comment (SargeKhan) [inline test/api/peers.js]: Why are we skipping peers/get tests now?

Comment (MaciejBaj) [reply test/api/peers.js]: quoting commit message: `skip part of peers tests relying on existing peer (fixed in #710)`

**Pull request merged**